pkg/queue/priority_queue_test.go

```diff
@@ -0,0 +1,142 @@
+package queue
```

Comment:
Can we add some examples here as well so that there's examples in the generated function docs?